extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/service/TransferProcessServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition.service;
+
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcess;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates;
+import org.eclipse.dataspaceconnector.transfer.core.command.commands.CancelTransferCommand;
+import org.eclipse.dataspaceconnector.transfer.core.command.commands.DeprovisionRequest;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.util.stream.Collectors.toList;
+
+public class TransferProcessServiceImpl implements TransferProcessService {
+    private final TransferProcessStore transferProcessStore;
+    private final TransferProcessManager manager;
+    private final TransactionContext transactionContext;
+
+    public TransferProcessServiceImpl(TransferProcessStore transferProcessStore, TransferProcessManager manager, TransactionContext transactionContext) {
+        this.transferProcessStore = transferProcessStore;
+        this.manager = manager;
+        this.transactionContext = transactionContext;
+    }
+
+    @Override
+    public TransferProcess findById(String transferProcessId) {
+        var result = new AtomicReference<TransferProcess>();
```

Comment:
What is the value of using AtomicReference in these methods?

Reply:
resolved by upstream PR